test/swar/BasicOperations.cpp

```diff
@@ -455,3 +456,36 @@ TEST_CASE(
     CHECK(SWAR<4, u16>(0x0F00).value() == saturatingUnsignedAddition(SWAR<4, u16>(0x0800), SWAR<4, u16>(0x0800)).value()); 
     CHECK(S4_32(0x0F0C'F000).value() == saturatingUnsignedAddition(S4_32(0x0804'F000), S4_32(0x0808'F000)).value()); 
 }
+
+TEST_CASE(
+    "cheapNullChecking",
+    "[swar][unsigned-swar]"
+) {
+    {
+        // The last value / lsb null lane is all these care about, but the
+        // implementation is so tied to the execution we check both.
+        CHECK(S4_32(0x0008'0080).value() == leastSignificantZeroLane<4, uint32_t>( S4_32(0xaaa0'aa0a)));
+        CHECK(1 == S4_32(leastSignificantZeroLane<4, uint32_t>( S4_32(0xaaa0'aa0a))).lsbIndex());
+        CHECK(S4_32(0x8888'8888).value() == leastSignificantZeroLane<4, uint32_t>( S4_32(0x1110'1110)));
+        CHECK(S4_32(0x8800'0000).value() == leastSignificantZeroLane<4, uint32_t>( S4_32(0x1025'3552)));
+        CHECK(S4_32(0x8880'0000).value() == leastSignificantZeroLane<4, uint32_t>( S4_32(0x1105'5352)));
```

Comment:
The formatting of these checks require work to verify.
Perhaps use normal indentation to facilitate the comprehension?